src/main/java/hyperledger/besu/java/rest/client/listener/SmartContractEventListener.java

```diff
@@ -2,28 +2,32 @@
 
 import hyperledger.besu.java.rest.client.config.EthConfig;
 import hyperledger.besu.java.rest.client.config.EthEventsProperties;
-import hyperledger.besu.java.rest.client.exception.ErrorCode;
-import hyperledger.besu.java.rest.client.exception.NotFoundException;
 import hyperledger.besu.java.rest.client.filters.TopicFilterHandler;
 import org.apache.commons.lang3.ObjectUtils;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Scope;
-import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
 
-@Component
-@Scope("prototype")
+@Service
 public class SmartContractEventListener implements Runnable {
+  /** Loads the eth-connector config. */
   @Autowired private EthConfig ethConfig;
+
+  /** Loads the topic-filter service. */
   @Autowired private TopicFilterHandler topicFilterHandler;
-  @Autowired private EthEventsProperties ethEventsProperties;
 
+  /** Loads the event-properties config. */
+  @Autowired(required = false)
+  private EthEventsProperties ethEventsProperties;
+
+  /**
+   * This run method will be called by AsyncEventListener
+   *
+   * <p>after successful Bean creation.
+   */
   @Override
   public void run() {
-    // only use smartContract address
-    if (ObjectUtils.isEmpty(ethEventsProperties.getSmartContract())
-        && ObjectUtils.isEmpty(ethEventsProperties.getSmartContract().getAddresses())) {
-      throw new NotFoundException(ErrorCode.NOT_FOUND, "Smart Contract Address list is empty");
+    if (ObjectUtils.isNotEmpty(EthEventsProperties.getEventNameByHash())) {
```

Comment:
Q: why we removed the prior logic for checking smart contract address ?

Reply:
Hashnames are formed when the smart contract & abi files are valid. So instead of checking them individually, i am directly checking for hashnames. That way, both the SC and abi will be validated